clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -268,6 +280,13 @@ public void poll(long timeout) throws IOException {
         currentTimeNanos = endSelect;
         this.sensors.selectTime.record(endSelect - startSelect, time.milliseconds());
 
+        // Perform setup for instantly connected channels since they won't be included in selectedKeys()
+        if (connectableChannels.size() > 0)
+            for (KafkaChannel channel : connectableChannels) {
+                sensors.maybeRegisterConnectionMetrics(channel.id());
```

Comment:
@llowrey : In the interest of time, @ijuma helped produce a new patch in https://github.com/apache/kafka/pull/1094. It would be great if you can verify if it fixes the problem. We will make sure that you get the acknowledgement when we commit the patch.